### PR TITLE
Modify spec to cover include_paths.

### PR DESCRIPTION
**Pull request by fhwang**

### CODE REVIEW

**Comment by GordonDiggs** [comment]
@mrb

File: SPEC.md [hunk 1]
```diff
@@ -18,9 +18,30 @@ Engines can define their own appropriate configuration keys and values, based on
...
 
-* `exclude_paths` -- An Array of file paths (relative to the `/code` directory) that should be ignored for the purposes of analysis. No Issues should be emitted for excluded files.
+* `include_paths`
+* `exclude_paths` (deprecated)
+
+These are covered in detail below.
+
+### Which files to analyze
+
+`include_paths` in `/config.json` contains an Array of file paths and directory paths (relative to the `/code` directory) that defines the range of files that the engine can analyze.  Directories will end with a trailing slash.  For example:
```
Comment (andyw8) [inline SPEC.md]: s/Array/array ?

File: SPEC.md [hunk 2]
```diff
@@ -18,9 +18,30 @@ Engines can define their own appropriate configuration keys and values, based on
...
+{
+  "include_paths":[
+    ".gitignore",
+    "app/",
+    "test/"
+  ]
+}
+```
+
+`include_paths` may include paths to files that are irrelevant to the analysis (i.e., `.gitignore` if the engine analyzes Javascript).  It is recommended that the engine filters out irrelevant files as opposed to passing them to the analysis library.
```
Comment (andyw8) [inline SPEC.md]: s/Javascript/JavaScript ?

**Comment by fhwang** [comment]
Good catches @andyw8 , thanks.

File: SPEC.md [hunk 3]
```diff
@@ -18,9 +18,30 @@ Engines can define their own appropriate configuration keys and values, based on
 their needs. A developer invoking a Code Climate engine stores their configuration
 for all engines in a single `.codeclimate.yml` file. The Code Climate CLI (and other tools compatible with Code Climate Engines) parse and interpret the YAML file to produce a single, simple JSON config object for each engine.
 
-Certain keys of the config object will be passed to all engines, and must be respected in order to conform with the specification. Today, there is only one such key:
+Certain keys of the config object will be passed to all engines, and must be respected in order to conform with the specification.  There are two keys, both related to which files to analyze:
```
Comment (pbrisbin) [inline SPEC.md]: :lipstick: one space between sentences.

File: SPEC.md [hunk 4]
```diff
@@ -18,9 +18,30 @@ Engines can define their own appropriate configuration keys and values, based on
...
 
-* `exclude_paths` -- An Array of file paths (relative to the `/code` directory) that should be ignored for the purposes of analysis. No Issues should be emitted for excluded files.
+* `include_paths`
+* `exclude_paths` (deprecated)
+
+These are covered in detail below.
+
+### Which files to analyze
+
+`include_paths` in `/config.json` contains an array of file paths and directory paths (relative to the `/code` directory) that defines the range of files that the engine can analyze.  Directories will end with a trailing slash.  For example:
```
Comment (pbrisbin) [inline SPEC.md]: Same here.

Comment (pbrisbin) [inline SPEC.md]: And everywhere...

**Comment by fhwang** [comment]
@mrb @brynary Can I get some review on this? This is the last PR blocking progress on the exclude-files issue.

File: SPEC.md [hunk 5]
```diff
@@ -18,9 +18,30 @@ Engines can define their own appropriate configuration keys and values, based on
...
+{
+  "include_paths":[
+    ".gitignore",
+    "app/",
+    "test/"
+  ]
+}
+```
+
+`include_paths` may include paths to files that are irrelevant to the analysis (i.e., `.gitignore` if the engine analyzes JavaScript). It is recommended that the engine filters out irrelevant files as opposed to passing them to the analysis library.
```
Comment (brynary) [inline SPEC.md]: I'd change the last sentence to just say: "Engines are responsible for filtering out irrelevant files."

**Comment by brynary** [comment]
Minor tweak but LGTM

File: SPEC.md (hunk 5 above)
Comment (mrb) [inline SPEC.md]: +1

**Comment by mrb** [comment]
:shipit:

**Pull request merged**